packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/core/Parser.java

```diff
@@ -482,4 +488,27 @@ public Collection<Plugin> getPlugins() {
         }
 
     }
+
+    public static class SpringBootAppRootPackageFinder {
+
+        public Optional<String> findRootPackage(List<String> buildDirectories) {
+
+            var currentModuleClassGraph = new ClassGraph()
+                    .enableAnnotationInfo().ignoreClassVisibility()
+                    .overrideClassLoaders(ClassLoader.getSystemClassLoader())
+                    .overrideClasspath(buildDirectories);
+
+            String springBootApplicationAnnotation = "org.springframework.boot.autoconfigure.SpringBootApplication";
+
+            try (var scanResult = currentModuleClassGraph.scan()) {
```

Comment:
Does scanning everything again have some performance impact?

Reply:
This is only looking into the `classes` directory and is not scanning the dependencies.